File: easy/455.assign-cookies.java
import java.util.Arrays;

/*
 * @lc app=leetcode id=455 lang=java
 *
 * [455] Assign Cookies
 */

// @lc code=start
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        // must sort, and give the less greedy child smaller cookie
        Arrays.sort(g);
        Arrays.sort(s);

        int count = 0;
        for (int i=0, j=0; i<g.length && j<s.length;) {
            if (g[i] <= s[j]) {
                // content, next one
                count++;
                i++;
                j++;
            } else {
                // try next cookie
                j++;
            }
        }

        return count;
    }
}
// @lc code=end

File: easy/455.assign-cookies.py
#
# @lc app=leetcode id=455 lang=python3
#
# [455] Assign Cookies
#

# @lc code=start
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()

        ans = 0
        cookie = 0
        for greed in g:
            while cookie < len(s) and greed > s[cookie]:
                cookie += 1
            
            if cookie == len(s):
                break

            cookie += 1
            ans += 1
        
        # [1,2,3]\n[3]
        return ans
# @lc code=end

